protocol/x/clob/keeper/orders.go

```diff
@@ -46,6 +46,83 @@ func (k Keeper) GetOperations(ctx sdk.Context) *types.MsgProposedOperations {
 	return msgProposedOperations
 }
 
+// BatchCancelShortTermOrder removes a specified batch of short term orders from all order-related data
+// structures in the memclob. As well, BatchCancelShortTermOrder adds (or updates) cancels to the desired
+// `goodTilBlock` in the memclob for all specified orders.
+// This message is not atomic. It will optimistically call `CancelShortTermOrder` for every order in the batch.
+// If any of the orders error, the error will be silently logged. This msg will only error if:
+// - Stateful validation fails
+// - Every cancel in the batch fails.
+// This method assumes the provided MsgBatchCancel has already passed ValidateBasic in CheckTx.
+func (k Keeper) BatchCancelShortTermOrder(
+	ctx sdk.Context,
+	msg *types.MsgBatchCancel,
+) error {
+	lib.AssertCheckTxMode(ctx)
+	// Note that we add `+1` here to account for the fact that `ctx.BlockHeight()` is technically the
+	// previously mined block, not the next block that will be proposed. This is due to the fact that
+	// this function is only ever called during `CheckTx`.
+	nextBlockHeight := lib.MustConvertIntegerToUint32(ctx.BlockHeight() + 1)
+
+	// Statefully validate the GTB and the clob pair ids.
+	goodTilBlock := msg.GetGoodTilBlock()
+	if err := k.validateGoodTilBlock(goodTilBlock, nextBlockHeight); err != nil {
+		return err
+	}
```

Comment:
don't we do this in `CancelShortTermOrder`?

Reply:
we do but i'd rather just error out here than have each one error out slowly in each CancelShortTermOrder call and then return the user a `BatchCancelAllFailed with error: invalid GTB`, a top level `invalid gtb` seems better

Reply:
then we can just call `k.MemClob.CancelOrder` directly? I guess not as future proof

Reply:
small nit: I kind of don't like this `+1` logic everywhere and feel like we should move it to within `validateGoodTilBlock`. 

haven't checked other references to `validateGoodTilBlock`, so will leave it to you to decide

Reply:
Hm we could call memclob directly. I'm ok with either.

Current approach
- `SendOffchainMessages` gets called multiple times for each 
- each cancel gets its own latency metric exposed
Calling memclob directly
- can coalesce offchain messages and call `SendOffchainMessages` once
- no latency metrics for each cancel

re: the +1 logic, i'm kind of for keeping it as it is since i don't want `validateGoodTilBlock` method to assume that it's being used in checktx. i'd prefer to keep it as a pure function of its parameters without assuming anything about usage

Reply:
> re: the +1 logic, i'm kind of for keeping it as it is since i don't want validateGoodTilBlock method to assume that it's being used in checktx. i'd prefer to keep it as a pure function of its parameters without assuming anything about usage

I meant do something like 
``` 
func validateGTB(ctx, ...) {
    if ctx.IsCheckTx() {
        GTB = ctx.BlockHeight() + 1
    }
    GTB = ctx.BlockHeight()
}
```

slightly cleaner imo, but up to you